elements/pl-units-input/pl-units-input.py

```diff
@@ -0,0 +1,468 @@
+            info = chevron.render(
+                f,
+                {"format": True, "only_units": grading_mode is GradingMode.ONLY_UNITS},
+            ).strip()
+
+        if pl.has_attrib(element, "placeholder"):
+            placeholder_text = pl.get_string_attrib(element, "placeholder")
+        elif grading_mode is GradingMode.ONLY_UNITS:
+            placeholder_text = "Unit"
+        elif grading_mode is GradingMode.WITH_UNITS:
+            atol = pl.get_string_attrib(element, "atol", ATOL_DEFAULT)
+            placeholder_text = f"Number (atol={atol}) + Unit"
+
+        # Grading mode must be units fixed at this point,
+        # now can case based on comparison
+        elif comparison is uu.ComparisonType.RELABS:
+            rtol = pl.get_float_attrib(element, "rtol", RTOL_DEFAULT)
+            atol = pl.get_string_attrib(element, "atol", ATOL_DEFAULT)
+            placeholder_text = f"Number (rtol={rtol}, atol={atol}) + Unit"
+        elif comparison is uu.ComparisonType.EXACT:
+            placeholder_text = "Number (exact) + Unit"
+        elif comparison is uu.ComparisonType.SIGFIG:
+            digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
+            figure_str = "figure" if digits == 1 else "figures"
+            placeholder_text = f"Number ({digits} significant {figure_str}) + Unit"
+        elif comparison is uu.ComparisonType.DECDIG:
+            digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
+            digit_str = "digit" if digits == 1 else "digits"
+            placeholder_text = f"Number ({digits} {digit_str} after decimal) + Unit"
+        else:
+            assert_never(comparison)
+
+        html_params = {
+            "question": True,
+            "name": name,
+            "label": label,
+            "suffix": suffix,
+            "editable": editable,
+            "info": info,
+            "size": size,
+            "show_info": show_info,
+            "placeholder": placeholder_text,
+            "uuid": pl.get_uuid(),
+            display.value: True,
+            "display_append_span": show_info or suffix,
+        }
+
+        if score is not None:
+            score_type, score_value = pl.determine_score_params(score)
+            html_params[score_type] = score_value
+
+        if raw_submitted_answer is not None:
+            html_params["raw_submitted_answer"] = escape(raw_submitted_answer)
+
+        with open(UNITS_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
+            return chevron.render(f, html_params).strip()
+
+    elif data["panel"] == "submission":
+        parse_error = data["format_errors"].get(name, None)
+        html_params = {
+            "submission": True,
+            "label": label,
+            "parse_error": parse_error,
+            "uuid": pl.get_uuid(),
+        }
+
+        if parse_error is None and name in data["submitted_answers"]:
+            # Get submitted answer, raising an exception if it does not exist
```

Comment:
this comment doesn't seem necessary

Reply:
Removed in [4d5b245](https://github.com/PrairieLearn/PrairieLearn/pull/6790/commits/4d5b245479e0260867e69994f65d9d5b8dd71cdb).